pkg/admission/errors.go

```diff
@@ -46,3 +47,21 @@ func NewForbidden(a Attributes, internalError error) error {
 	}
 	return apierrors.NewForbidden(kind, name, internalError)
 }
+
+func NewNotFound(a Attributes) error {
```

Comment:
There is a lot of common code between this and NewForbidden to determine the name for a resource.  Can you make a utility and share it between the functions?